examples/C++/DDS/HelloWorldExampleDiscoveryServer/arg_configuration.h

```diff
@@ -0,0 +1,178 @@
+    static option::ArgStatus String(
+            const option::Option& option,
+            bool msg)
+    {
+        if (option.arg != 0)
+        {
+            return option::ARG_OK;
+        }
+        if (msg)
+        {
+            print_error("Option '", option, "' requires a string argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static option::ArgStatus Locator(
+            const option::Option& option,
+            bool msg)
+    {
+        if (option.arg != 0)
+        {
+            // we must check if it is a correct ip address plus port number
+            if (std::regex_match(option.arg, ipv4))
+            {
+                return option::ARG_OK;
+            }
+        }
+        if (msg)
+        {
+            print_error("Option '", option, "' requires an IPaddress[:portnumber] argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static const std::regex ipv4;
+};
+
+enum  optionIndex
+{
+    UNKNOWN_OPT,
+    HELP,
+    TOPIC,
+    WAIT,
+    SAMPLES,
+    INTERVAL,
+    LOCATOR
+};
+
+const option::Descriptor usage[] = {
+    { UNKNOWN_OPT, 0, "", "",                Arg::None,
+      "Usage: HelloWorldExampleDiscoveryServer <publisher|subscriber|server>\n\nGeneral options:" },
+    { HELP,    0, "h", "help",               Arg::None,      "  -h \t--help  \tProduce help message." },
+
+    { UNKNOWN_OPT, 0, "", "",                Arg::None,      "\nPublisher options:"},
+    { TOPIC, 0, "t", "topic",                  Arg::String,
+      "  -t <topic_name> \t--topic=<topic_name>  \tTopic name (Default: HelloWorldTopic)." },
+    { WAIT, 0, "w", "wait",                 Arg::Numeric,
+      "  -w <num> \t--wait=<num> \tNumber of matched subscribers required to publish"
+      " (Default: 0 => does not wait)." },
+    { SAMPLES, 0, "s", "samples",              Arg::Numeric,
+      "  -s <num> \t--samples=<num>  \tNumber of samples to send (Default: 0 => infinite samples)." },
+    { INTERVAL, 0, "i", "interval",            Arg::Numeric,
+      "  -i <num> \t--interval=<num>  \tTime between samples in milliseconds (Default: 100)." },
+    { LOCATOR, 0, "", "ip",                 Arg::Locator,
+      "  \t--ip=<IPaddress[:port number]>  \tServer address (Default address: 127.0.0.1, default port: 60006)." },
+
+    { UNKNOWN_OPT, 0, "", "",                Arg::None,      "\nSubscriber options:"},
+    { TOPIC, 0, "t", "topic",                  Arg::String,
+      "  -t <topic_name> \t--topic=<topic_name>  \tTopic name (Default: HelloWorldTopic)." },
+    { SAMPLES, 0, "s", "samples",              Arg::Numeric,
+      "  -s <num> \t--samples=<num>  \tNumber of samples to wait for (Default: 0 => infinite samples)." },
+    { LOCATOR, 0, "", "ip",                 Arg::Locator,
+      "  \t--ip=<IPaddress[:port number]>  \tServer address (Default address: 127.0.0.1, default port: 60006)." },
+
+    { UNKNOWN_OPT, 0, "", "",                Arg::None,      "\nDiscoveryServer options:"},
```

Comment:
It is not clear if this parameter is to set the listening address or the connection address.
I would use `ip` as connection (as in pub & sub) and use a new parameter to set its own locators